app/db.py

```diff
@@ -60,63 +63,76 @@ class OHLCV(Base):
 
 def symbol_exists(symbol):
     """ check if the stock symbol exists in the database """
-    return session.query(exists().where(OHLCV.symbol==symbol))[0]
+    with session_scope() as session:
+        return session.query(
+            exists()
+            .where(
+                OHLCV.symbol==symbol,
+            )
+        )[0][0]
+
 
 
 def get_recent_ohlvc(symbol):
     """ get the last 10 days of data for a stock symbol """
-    lastTenDays = []
+    last_ten_days = []
     with session_scope() as session:
-        data = session.query(OHLCV). \
-            filter_by(symbol=symbol). \
-            order_by(OHLCV.date.desc()). \
-            limit(10)
-
-    for day in data:
+        result = session.query(
+            OHLCV,
+        ).filter_by(
+            symbol=symbol,
+        ).order_by(
+            OHLCV.date.desc(),
+        ).limit(
+            10,
+        )
+
+    for day in result:
         dayDict = dict(
             date=day.date,
             symbol=day.symbol,
             open=day.open,
             high=day.high,
             low=day.low,
             close=day.close,
-            volume=day.volume
-            )
-        lastTenDays.append(dayDict)
+            volume=day.volume,
+        )
+        last_ten_days.append(dayDict)
 
-    return lastTenDays
+    return last_ten_days
 
 
 def get_recent_data_date(symbol):
     """ get the most recent date that we have daily stock data """
-    start_date = ''
     with session_scope() as session:
-        for date in session.query(func.max(OHLCV.date)). \
-                            filter_by(symbol=symbol):
-            if date[0]:
-                start_date = date[0]
-    return start_date
+        date, = session.query(
+            func.max(OHLCV.date),
+        ).filter_by(
+            symbol=symbol,
+        ).one_or_none()
+        return date
 
 
 def save_stock_data(data, symbol):
     """ save daily stock data """
     with session_scope() as session:
         for record in data.itertuples():
             session.add(OHLCV(symbol=symbol,
-                date=record[0],
-                open=record[1],
-                high=record[2],
-                low=record[3],
-                close=record[4],
-                volume=record[5]))
+                    date=record[0],
+                    open=record[1],
+                    high=record[2],
+                    low=record[3],
+                    close=record[4],
+                    volume=record[5],
+                )
+            )
```

Comment:
I didn't notice the first keyword argument right away because it's at different level of indentation than the rest of the keyword arguments.  It's correct, but I had to read it a couple times before I realized I was making it more complicated than it is.

Although it's worth mentioning that just because I had trouble reading a code block doesn't mean that it should be changed.  It could just mean I was expecting something else, I was distracted when I read it, etc.